### PR TITLE
Enhance SoftmaxWithLoss by adding select_label.

### PR DESCRIPTION
**Pull request by SOLARleisu**
Selecting only a specified class for backprop using SoftmaxWithLoss.

1. Add an new variable for softmax_loss_layer to allow the user to give
   different weights to different classes (or labels).
   use select_label like:
   loss_param {
     select_label: 0
     }
2. Tested with examples/mnist/train_lenet.sh for both GPU mode and CPU
   mode.
3. Use examples/mnist/lenet_train_test.prototxt as an example for
   select_label (has been commented out).
4. Note: ignore_label gets the priority if select_label and ignore_label
   were in conflict.

### CODE REVIEW

**Comment by SOLARleisu** [comment]
@Noiredd 
Thank you for you advice for "Selecting only a specified class for backprop using SoftmaxWithLoss #6253".
I push this new request for the same work.

**Comment by SOLARleisu** [comment]
@Noiredd 
I finally choose "select_label_" in the c++ code and use "select_label" in protobuffer file.

**Comment by Noiredd** [comment]
This looks good so far. Have you considered adding some tests to this feature? See [TestForwardIgnoreLabel](https://github.com/BVLC/caffe/blob/300f43f3ae6347ac8e01093f9a57ee99e551ed74/src/caffe/test/test_softmax_with_loss_layer.cpp#L64) for reference.

**Comment by SOLARleisu** [comment]
@Noiredd 
Ok. I will give the tests related code. Your advice is really useful.
Thank you very much.
